autopush-common/src/util/user_agent.rs

```diff
@@ -55,6 +57,16 @@ impl From<&str> for UserAgentInfo {
     }
 }
 
+impl From<&actix_web::HttpRequest> for UserAgentInfo {
+    fn from(req: &actix_web::HttpRequest) -> UserAgentInfo {
+        if let Some(header) = req.headers().get(&actix_web::http::header::USER_AGENT) {
+            Self::from(header.to_str().unwrap_or("UNKNOWN"))
+        } else {
+            UserAgentInfo::default()
```

Comment:
This will just create an empty default `UserAngentInfo`. Is there any desire to implement a `default` impl to fine grain those values beyond their empty data types?

Reply:
Good point, we should probably use `Self::from("UNKNOWN")` (which I believe causes woothee to fill in all values with "UNKNOWN"?). Maybe that should be the default instead of the empty strings which doesn't seem useful for the typical usage of this (metrics, sentry tags, etc.)

Reply:
..and we could punt that change to another issue

Reply:
Yeah, let's punt on switching to "UNKNOWN" as the default for UserAgentInfo. ~I'll file a quick ticket for it.~ see https://mozilla-hub.atlassian.net/browse/SYNC-4355